### PR TITLE
fix: merge api results after their transformations (TCTC-9634)

### PR DESCRIPTION
**Pull request by julien-pinchelimouroux**
## Why

API results where appended before any user's configured transformations (filters, flatten). We have to append resulted dataframes.
Moreover, APIs can return directly json objects, which created invalid results merging.
In addition, we have to add a data_length_filter to offset_limit pagination in case of the data result isn't relevant to compute the data length.

### CODE REVIEW

**Review by davinov — APPROVED** [review]
Indeed, thanks for this fix and the associated test!

File: toucan_connectors/http_api/pagination_configs.py [hunk 1]
```diff
@@ -56,6 +56,13 @@ class OffsetLimitPaginationConfig(PaginationConfig):
     offset: int = Field(0, **UI_HIDDEN)
     limit_name: str = "limit"
     limit: int
+    data_length_filter: str = Field(
+        ".",
+        description=(
+            "Filter to determine the data length and compare it to the limit value. "
+            "It must point to a list of results. " + FilterSchemaDescription
```
Comment (lukapeschke) [inline toucan_connectors/http_api/pagination_configs.py:63]:
```suggestion
            "Filter to access the received data. Allows to compare its length to the limit value. "
            "It must point to a list of results. " + FilterSchemaDescription
```

Wouldn't `data_filter` be a better name ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Fix
+
+- HTTP API: Extract API results dataframes separately and merge them after.
+- HTTP API: Add `data_length_filter` offset pagination config field to determine which part of data must be used to compute the data length.
```
Comment (lukapeschke) [inline CHANGELOG.md:8]: Shouldn't this be in the `Added` section ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Fix
+
+- HTTP API: Extract API results dataframes separately and merge them after.
```
Comment (lukapeschke) [inline CHANGELOG.md:7]: The changelog is destined to end-users. Instead of explaining the code change, could you explain the issue that was solved ?

**Review by lukapeschke — CHANGES_REQUESTED** [review]
Thanks! few nits here and there. And `pandas.concat` should be compatible with v1 and v2, AND do what you want :slightly_smiling_face:

**Review by lukapeschke — APPROVED** [review]
thx!

**Pull request merged**